scripts/deploy.sh

```diff
@@ -31,14 +31,22 @@ cd ${OA_DIR}
 if [[ "${DEPLOY_AIO}" == "yes" ]]; then
   # force the deployment of haproxy for an AIO
   export DEPLOY_HAPROXY="yes"
-  if [[ ! -d /etc/openstack_deploy/ ]]; then
+  if [[ ! -e /etc/openstack_deploy/openstack_user_config.yml ]]; then
     ./scripts/bootstrap-aio.sh
     pushd ${RPCD_DIR}
-      for filename in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
-        if [[ ! -a "/${filename}" ]]; then
-          cp "${filename}" "/${filename}";
+      for relative_path in $(find etc/openstack_deploy/ -type f -iname '*.yml'); do
+        # If the relative_path exists in /etc/openstack_deploy then it is merged
+        # into then repo version overriding the repo value when there is a
+        # conflict.
+        if [[ -e "/${relative_path}" ]]; then
+          # Output is written to the base file
+          #                         Base                 Override
+          ../scripts/update-yaml.py "${relative_path}"   "/${relative_path}";
+          #                         repo/rpcd/etc        /etc/openstack_deploy
         fi
-        ../scripts/update-yaml.py "/${filename}" "${filename}";
+          # Copy (possibly updated) config files from repo to
+          # /etc/openstack_deploy
+          cp "${relative_path}" "/${relative_path}";
```

Comment:
This dirties up the repo meaning a it will need to be cleaned up before pulling down again.